cloudstack/loadbalancer.go

```diff
@@ -843,25 +844,37 @@ func sortPorts(ports []v1.ServicePort) {
 	})
 }
 
-func comparePorts(ports []v1.ServicePort, rule *loadBalancerRule) bool {
+func comparePorts(service *v1.Service, rule *loadBalancerRule) bool {
+	ports := service.Spec.Ports
+	_, useTargetPort := getLabelOrAnnotation(service.ObjectMeta, lbUseTargetPort)
 	sortPorts(ports)
 	var additionalPorts []string
 	for _, p := range ports[1:] {
-		additionalPorts = append(additionalPorts, fmt.Sprintf("%d:%d", p.Port, p.NodePort))
+		if useTargetPort {
+			additionalPorts = append(additionalPorts, fmt.Sprintf("%d:%d", p.Port, p.TargetPort.IntVal))
```

Comment:
Use the `IntValue()` method instead of the `IntVal` field as it automatically handle conversions if needed.